edgecontext.thrift

```diff
@@ -104,6 +118,24 @@ struct RequestId {
     1: string readable_id
 }
 
+/** Locale data from a request to our services that we want to
+propagate between services.
+
+This model is a component of the "Edge-Request" header.  You should not need to
+interact with this model directly, but rather through the EdgeRequestContext
+interface provided by baseplate.
+
+*/
+struct Locale {
+    /** Language code representing the client language preferences.
+    */
+    1: LanguageCode language_code
+
+    /** Region code representing the client configured locale region.
+    */
+    2: RegionCode region_code
```

Comment:
🔕  Do we really want language code and region code to be separated fields? I think conventionally we just use a single string (e.g. `en_US`) for the locale in most cases.

If the concern is that in some cases we would only have language code without region code (e.g. we only have `en` but don't know which variant of en that is), then region code should be optional?

Reply:
Yeah, good point. I've gone back and forth on this a bit, but perhaps combining them into a single field is the way to go.

I figured that having them split up might make it slightly cleaner to handle things like falling back to a non-region-specific language if a regional variant isn't available. e.g. `en-GB` isn't available so fallback to `en`, but it's simple enough to parse that out from a single string, and saves having to extract them in the first place. Not to mention most i18n libraries can negotiate that by default. 

It should also make the distinction between the region and country codes clearer. Will update.

Reply:
Agreed. If we ever need to access the things independently, not having to do parsing in every client (even if it's just splitting on `_`) is pretty nice. But if the vast majority of use cases are passing the full `{lang}_{region}` thing into an i18n library (meaning we're pushing the `_` joining into every client)  then we might just want to send that string directly. 

@ZackMcBride do you have an idea for how this will generally be used?

Reply:
Yeah I think the simplest way to go here is to provide the combined string. For the most part we will just be passing this through to a library (like [Babel](http://babel.pocoo.org/en/latest/api/core.html?highlight=negotiate#babel.core.negotiate_locale), which accepts `xx_XX`) to negotiate the locale, so the combined one is gonna be less work overall. I think any manual fallbacks which require parsing each component will probably be the exception. Gonna update, thanks folks.

Reply:
Updated in https://github.com/reddit/edgecontext/pull/12/commits/371b0481603ed30f4874e03f7233b340c657fd95